wasm/store.go

```diff
@@ -40,59 +81,119 @@ type (
+		// FunctionType holds the pointer to TypeInstance whose functionType field equals that of this function.
+		FunctionType *TypeInstance
+		// NumLocals holds the number of locals in this function.
+		NumLocals uint32
+		// LocalTypes holds types of locals.
+		LocalTypes []ValueType
+		// HostFunction holds the runtime representation of host functions.
+		// If this is not nil, all the above fields are ignored as they are specific to non-host functions.
+		HostFunction *reflect.Value
+		// Address is the funcaddr(https://www.w3.org/TR/wasm-core-1/#syntax-funcaddr) of this function insntance.
+		// More precisely, this equals the index of this function instance in store.FunctionInstances.
+		// All function calls are made via funcaddr at runtime, not the index (scoped to a module).
+		//
+		// This is used by both host and non-host functions.
+		Address FunctionAddress
+		// Name is for debugging purpose, and is used to argument the stack traces.
+		//
+		// For non-host functions, this holds the value of
+		// name for this function which might be stored in the "name" custom section.
+		// Otherwise, the value is set to "unknown".
+		//
+		// For host functions, this is in the form of "{moduleName}.{funcName}" where these params
+		// are given via storeAddHostFunction.
+		Name string
+	}
+
+	// TypeInstance is a store-specific representation of FunctionType where the function type
+	// is coupled with TypeID which is specific in a store.
+	TypeInstance struct {
+		Type *FunctionType
+		// TypeID is assigned by a store for FunctionType.
+		TypeID FunctionTypeID
+	}
+
+	// GlobalInstance represents a global instance in a store.
+	// See https://www.w3.org/TR/wasm-core-1/#global-instances%E2%91%A0
 	GlobalInstance struct {
 		Type *GlobalType
-		Val  uint64
+		// Val holds a 64-bit representation of the actual value.
+		Val uint64
 	}
 
+	// TableInstance represents a table instance in a store.
+	// See https://www.w3.org/TR/wasm-core-1/#table-instances%E2%91%A0
+	//
+	// Note this is fixed to function type until post MVP reference type is implemented.
 	TableInstance struct {
-		Table    []*TableInstanceElm
-		Min      uint32
-		Max      *uint32
+		// Table holds the table elements managed by this table instance.
+		//
+		// Note: we intentionally use "[]TableElement", not "[]*TableElement",
+		// as JIT engine access this slice directly from assembly.
+		// If pointer type is used, the access becomes two level indirection (two hops of pointer jumps)
+		// which is a bit costly. TableElement is 128 bit (two 64bit fields) so the cost of using value type
+		// would be ignorable.
+		Table []TableElement
+		Min   uint32
+		Max   *uint32
+		// Currently fixed to 0x70 (funcref type).
 		ElemType byte
 	}
 
-	TableInstanceElm struct {
-		Function *FunctionInstance
-		// Other ref types will be added.
+	// TableElement represents an item in a table instance.
+	//
+	// Note this is fixed to function type until post MVP reference type is implemented.
```

Comment:
Let's focus on the standard, as we do that elsewhere and if we update to a later standard using a string search is easier when consistent.

```suggestion
	// Note: this is fixed to function type as it is the only supported type in WebAssembly 1.0 (MVP)
```